Documentation/teaching/info/vm.rst

```diff
@@ -4,118 +4,118 @@
-		-net nic,model=virtio,vlan=0 -net tap,ifname=tap0,vlan=0,script=no,downscript=no\
-		-drive file=rootfs.img,if=virtio,format=raw --append "root=/dev/vda console=hvc0" \
-		--display none -s
-   char device redirected to /dev/pts/19 (label virtiocon0)
+   .../linux/tools/labs$ make boot
+
+The first run of the ``make boot`` command will compile the kernel image and it
+will take longer. Subsequent runs will only start the QEMU virtual machine,
+with verbose output provided:
 
+.. code-block:: shell
 
-.. note:: To show the qemu console use "QEMU_DISPLAY=sdl make
-          boot". This will show the VGA output and will also give
+   .../linux/tools/labs$ make boot
+   mkdir /tmp/tmp.7rWv63E9Wf
+   sudo mount -t ext4 -o loop core-image-minimal-qemux86.ext4 /tmp/tmp.7rWv63E9Wf
+   sudo make -C /home/razvan/school/so2/linux.git modules_install INSTALL_MOD_PATH=/tmp/tmp.7rWv63E9Wf
+   make: Entering directory '/home/razvan/school/so2/linux.git'
+     INSTALL crypto/crypto_engine.ko
+     INSTALL drivers/crypto/virtio/virtio_crypto.ko
+     INSTALL drivers/net/netconsole.ko
+     DEPMOD  4.19.0+
+   make: Leaving directory '/home/razvan/school/so2/linux.git'
+   sudo umount /tmp/tmp.7rWv63E9Wf
+   rmdir /tmp/tmp.7rWv63E9Wf
+   sleep 1 && touch .modinst
+   qemu/create_net.sh tap0
+
+   dnsmasq: failed to create listening socket for 172.213.0.1: Address already in use
+   qemu/create_net.sh tap1
+
+   dnsmasq: failed to create listening socket for 127.0.0.1: Address already in use
+   /home/razvan/school/so2/linux.git/tools/labs/templates/assignments/6-e100/nttcp -v -i &
+   nttcp-l: nttcp, version 1.47
+   nttcp-l: running in inetd mode on port 5037 - ignoring options beside -v and -p
+   bind: Address already in use
+   nttcp-l: service-socket: bind:: Address already in use, errno=98
+   ARCH=x86 qemu/qemu.sh -kernel /home/razvan/school/so2/linux.git/arch/x86/boot/bzImage -device virtio-serial -chardev pty,id=virtiocon0 -device virtconsole,chardev=virtiocon0 -serial pipe:pipe1 -serial pipe:pipe2 -netdev tap,id=tap0,ifname=tap0,script=no,downscript=no -net nic,netdev=tap0,model=virtio -netdev tap,id=tap1,ifname=tap1,script=no,downscript=no -net nic,netdev=tap1,model=i82559er -drive file=core-image-minimal-qemux86.ext4,if=virtio,format=raw -drive file=disk1.img,if=virtio,format=raw -drive file=disk2.img,if=virtio,format=raw --append "root=/dev/vda loglevel=15 console=hvc0" --display none -s
+   qemu-system-i386: -chardev pty,id=virtiocon0: char device redirected to /dev/pts/68 (label virtiocon0)
+
+.. note:: To show the QEMU console use ``QEMU_DISPLAY=sdl make
+          boot```. This will show the VGA output and will also give
           access to the standard keyboard.
 
+.. note:: The virtual machine setup scripts and configuration files are located
+          in ``tools/labs/qemu/``.
+
+.. note:: The kernel image is only built the first time the virtual machine is
+          started. It will be rebuilt each time there are updates to local
```

Comment:
This is not true, one needs to remove arch/x86/boot/bzImage to force trigger a new kernel rebuild

Reply:
Hmmm, I'll test that and update the document accordingly.